sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -468,6 +498,27 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
         return SqlParts.partsForSql(where, parameterArray);
     }
 
+    private static String buildWhereClauseArgument(List<Object> values,
+                                                   List<Object> sqlParameters) {
+        List<String> inOperands = new ArrayList<String>();
+        for (Object value : values) {
+            if (validatePredicateValue(value)) {
+                inOperands.add("?");
+                sqlParameters.add(String.valueOf(value));
+            } else {
+                logger.log(Level.SEVERE, "Predicate value is invalid.");
+                return null;
+            }
+        }
```

Comment:
This is actually redundant validation as it already exists in the validator.  `validateClause` https://github.com/cloudant/sync-android/blob/44270-query-in-nin-operator-support/sync-core/src/main/java/com/cloudant/sync/query/QueryValidator.java#L409 in the validator validates each clause which includes predicate validation using the `validatePredicateValue` method.  This validation in the translator has been removed in 5b5040e6a64ad6720ef17ace362f2020397d0880.